src/engraving/dom/note.cpp

```diff
@@ -3545,9 +3547,21 @@ const Note* Note::lastTiedNote() const
         if (std::find(notes.begin(), notes.end(), note->tieFor()->endNote()) != notes.end()) {
             break;
         }
-        if (!note->tieFor()->endNote()) {
+        auto next = note->tieFor()->endNote();
+        if (!next) {
             break;
         }
+        if (positionTickOffset > 0) {
+            auto chord = next->chord();
+            auto intervals = score()->spannerMap().findOverlapping(next->tick().ticks(), (next->tick() + chord->actualTicks()).ticks());
+            // find which volta current positionTickOffset matches
+            auto it = std::find_if(intervals.cbegin(), intervals.cend(), [positionTickOffset](const interval_tree::Interval<Spanner*>& interval) {
+                return interval.stop == positionTickOffset && interval.value != nullptr && interval.value->isVolta();
+            });
```

Comment:
I don't think this logic is quite right. Ideally the parameter passed into the 'lastTiedNote' would specifying what the volta/repeat # was.  

Also, this doesn't give any facility to allow a secondary volta (2nd time ending etc.) to specify a "tie continuation".  In fact MuseScore itself doesn't seem to allow this as far as I can tell?
Really the proper solution for that would allow actually creating multiple ties from the start note to notes in the various voltas, but that's a much bigger change.